### PR TITLE
FAI-14157 | Azure pipelines improvements (builds stream)

### PR DESCRIPTION
**Pull request by dbruno21**
## Description

Azure pipelines improvements (builds stream)
- Use finishedTime instead of queueTime as cursor. Using queueTime would led to missing data for builds that were synced before being finished. Updated getBuilds to sort by finishTime (builds not finished won't be returned)
- Stop writing build.number to avoid confusion (we write it to build.name anyway)
- Jobs / steps with status succeededWithIssues were being written as Queued (???). Updated to be Success.
- Build / BuildStep statuses switch mappings were wrong for values with uppercases (lowercased vs camelCase)

## Type of change
- [X] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts [hunk 1]
```diff
@@ -184,9 +184,8 @@ export abstract class AzurePipelineConverter extends Converter {
       case 'skipped':
         return {category: BuildStateCategory.Failed, detail};
       case 'succeeded':
-        return {category: BuildStateCategory.Success, detail};
       case 'succeededWithIssues':
```
[the remark this answers is not in the document]
Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts:187]: image

[the remark this answers is not in the document]
Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts:187]: I think you're getting confused with the diff (happened to me as well just now). Expand the lines above and check for #taskresult docs on line 178

[the remark this answers is not in the document]
Comment (ypc-faros) [reply destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts:187]: Nvm

Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts:187]: `detail` is `state.toLowerCase` so we need to lowercase `succeededWithIssues` in the switch

File: destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts [hunk 2]
```diff
@@ -75,14 +75,14 @@ export abstract class AzurePipelineConverter extends Converter {
...
+  convertBuildState(result: string | undefined): {
     category: string;
     detail: string;
   } {
-    if (!state) {
+    if (!result) {
       return {category: BuildStateCategory.Unknown, detail: 'undefined'};
     }
-    const detail = state.toLowerCase();
+    const detail = result.toLowerCase();
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts:85]: Need to lower `partiallySucceeded` in the switch

Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts]: nice catch. I'll just remove the lowerCase(), I think it's more readable the other way and think we can trust the API values

File: destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts (hunk 1 above)
Comment (dbruno21) [reply destinations/airbyte-faros-destination/src/converters/azurepipeline/common.ts:184]: same as https://github.com/faros-ai/airbyte-connectors/pull/1797#discussion_r1857490339

**Review by cjwooo — APPROVED** [review]

**Pull request merged**